includes/apple-exporter/builders/class-components.php

```diff
@@ -426,10 +457,69 @@ private function clean_up_components( &$component ) {
 			return;
 		}
 
+		// Convert HTML IDs to identifiers.
+		$component = $this->convert_ids_to_identifiers( $component );
+
 		// Trim the fat.
 		$component['text'] = trim( $component['text'] );
 	}
 
+	/**
+	 * Convert the 'id' attributes in the HTML text of a
+	 * component to unique identifiers to support internal
+	 * anchor links.
+	 *
+	 * @param Component $component The component whose 'id' attributes will be converted.
+	 *
+	 * @return Component The component with converted 'id' attributes.
+	 * @since 2.4.4
+	 *
+	 * @access private
+	 */
+	private function convert_ids_to_identifiers( &$component ) {
+		// Dictionary to hold identifiers as keys with value the number of times each is found.
+		$identifiers = [];
+
+		// Searching for 'id' in the HTML and removing the attribute
+		// and store (valid) ones as 'identifier' on the component.
+		$component['text'] = preg_replace_callback(
+			'/\bid=["\'](.*?)["\']/',
+			function ( $matches ) use ( &$component, &$identifiers ) {
+				// If 'id' starts with a digit, it's skipped,
+				// as it's not a valid identifier and Apple News
+				// will reject it.
+				if ( preg_match( '/^\d/', $matches[1] ) ) {
+					return '';
+				}
+
+				// Saving the 'id' as the 'identifier'.
+				$identifier = $matches[1];
+
+				// If this identifier already exists skip it (is a duplicate).
+				if ( isset( $identifiers[ $identifier ] ) ) {
+					return '';
```

Comment:
Good call, originally I was going to modify the identifiers to be name-1, name-2, but realized that didn't make sense since people wouldn't be able to target them in WordPress. I'll remove.